packages/playground/src/components/repo-import-form-component.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useEffect } from 'react'
+import {
+  Button,
+  ButtonGroup,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Checkbox
+} from '@harnessio/canary'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { FormFieldSet } from '../index'
+
+const importRepoFormSchema = z.object({
+  identifier: z.string(),
+  description: z.string(),
+  pipelines: z.boolean().optional(),
+  authorization: z.boolean().optional(),
+  provider: z.string().min(1, { message: 'Please select a provider' }),
+  password: z.string().optional(),
+  organization: z.string().min(1, { message: 'Please select an organization' }),
+  repository: z.string().min(1, { message: 'Please select a repository' })
+})
+
+export type ImportRepoFormType = z.infer<typeof importRepoFormSchema>
+
+export function RepoImportForm({
+  isLoading = false,
+  onCancel,
+  onSubmit
+}: {
+  isLoading?: boolean
+  onCancel: () => void
+  onSubmit: (data: ImportRepoFormType) => void
+}) {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    reset,
+    formState: { errors }
+  } = useForm<ImportRepoFormType>({
```

Comment:
@sans-harness , can we use `useZodForm` instead? So that we can make it consistent in all places.